.github/go-script/cmd/validator/validator.go

```diff
@@ -0,0 +1,222 @@
+package validator
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"github.com/coinhall/yacar/internal/config"
+	"github.com/coinhall/yacar/internal/enums"
+	"github.com/coinhall/yacar/internal/reader"
+	"github.com/coinhall/yacar_util"
+)
+
+func Start() {
+	projRoot := config.GetRootDir()
+
+	// Get JSONs to validate and load into memory
+	chainDirsGlob := strings.Join(enums.GetAllChainNames(), ",")
+	fileNamesGlob := strings.Join(enums.GetAllFileNames(), ",")
+	fileGlobPattern := reader.AsGlobPattern(projRoot, chainDirsGlob, fileNamesGlob)
+	filePaths, err := filepath.Glob(fileGlobPattern)
+	if err != nil {
+		log.Fatalf("error while getting file paths from glob pattern: %s", err)
+	}
+
+	// Validate JSONs
+	if err := validateYacarJSONs(filePaths); err != nil {
+		log.Fatal(err)
+	}
+
+	log.Printf("Validated the following files: %s", strings.Join(filePaths, "\n  "))
+}
+
+func validateYacarJSONs(filePaths []string) error {
+	var wg sync.WaitGroup
+	for _, filePath := range filePaths {
+		wg.Add(1)
+		go func(filePath string) error {
+			defer wg.Done()
+			if err := validateYacarJSON(filePath); err != nil {
+				return err
+			}
+			return nil
+		}(filePath)
+	}
+	wg.Wait()
+
+	return nil
+}
+
+func validateYacarJSON(filePath string) error {
+	// Load file into memory
+	file, err := os.Open(filePath)
+	if err != nil {
+		return fmt.Errorf("error while opening file: %s", err)
+	}
+	defer file.Close()
+
+	// Validate JSON
+	if err := validation_handler(filePath, file); err != nil {
+		return fmt.Errorf("error while validating JSON: %s", err)
+	}
+
+	return nil
+}
+
+// TODO: Use validator pkg or shift the respective logic into yacar_util pkg
```

Comment:
Go's validator pkg requires the `validate` tag on the struct that needs validation which isn't too bad, but I don't think its possible to extract what was violated to help with debugging. This is currently being done with the TS version of the CI.